sqlx-core/src/postgres/listen.rs

```diff
@@ -0,0 +1,188 @@
+use futures_core::future::BoxFuture;
+use futures_core::stream::BoxStream;
+
+use crate::connection::Connection;
+use crate::describe::Describe;
+use crate::executor::Executor;
+use crate::pool::PoolConnection;
+use crate::postgres::protocol::{Message, NotificationResponse};
+use crate::postgres::{PgArguments, PgConnection, PgPool, PgRow, Postgres};
+use crate::Result;
+
+type PgPoolConnection = PoolConnection<PgConnection>;
+
+impl PgConnection {
+    /// Register this connection as a listener on the specified channel.
+    ///
+    /// If an error is returned here, the connection will be dropped.
+    pub async fn listen(mut self, channel: &impl AsRef<str>) -> Result<PgListener<Self>> {
+        let cmd = format!(r#"LISTEN "{}""#, channel.as_ref());
+        let _ = self.execute(cmd.as_str(), Default::default()).await?;
+        Ok(PgListener::new(self))
+    }
+
+    /// Register this connection as a listener on all of the specified channels.
+    ///
+    /// If an error is returned here, the connection will be dropped.
+    pub async fn listen_all(
+        mut self,
+        channels: impl IntoIterator<Item = impl AsRef<str>>,
+    ) -> Result<PgListener<Self>> {
+        for channel in channels {
+            let cmd = format!(r#"LISTEN "{}""#, channel.as_ref());
+            let _ = self.execute(cmd.as_str(), Default::default()).await?;
+        }
+        Ok(PgListener::new(self))
+    }
+
+    /// Build a LISTEN query based on the given channel input.
+    fn build_listen_query(channel: &impl AsRef<str>) -> String {
+        format!(r#"LISTEN "{}";"#, channel.as_ref())
+    }
+}
+
+impl PgPool {
+    /// Fetch a new connection from the pool and register it as a listener on the specified channel.
+    pub async fn listen(&self, channel: &impl AsRef<str>) -> Result<PgListener<PgPoolConnection>> {
+        let mut conn = self.acquire().await?;
+        let cmd = PgConnection::build_listen_query(channel);
+        let _ = conn.execute(cmd.as_str(), Default::default()).await?;
```

Comment:
`.execute()` isn't the right call here because `LISTEN` isn't a preparable statement. Use `.send()` instead (which also doesn't require constructing a `PgArguments`).

Reply:
(Same thing with all the other usages of `.execute()`.)

Reply:
Sounds good.